diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index 4fa49b9..e58ca61 100755
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -2192,7 +2192,11 @@ public class CameraActivity extends Activity
                 return;
             }
         }
-        CameraHolder.instance().keep();
+        // A video mode transition needs a fully closed stock NVIDIA sensor.
+        if (moduleIndex != ModuleSwitcher.VIDEO_MODULE_INDEX &&
+                mCurrentModuleIndex != ModuleSwitcher.VIDEO_MODULE_INDEX) {
+            CameraHolder.instance().keep();
+        }
         closeModule(mCurrentModule);
         setModuleFromIndex(moduleIndex);
 
